src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -239,27 +239,10 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                     {
                         ssize_t ival = op3->AsIntCon()->IconValue();
                         assert((ival >= 0) && (ival <= 255));
-
-                        if ((intrinsicId == NI_SSE41_Insert) && (baseType == TYP_FLOAT))
```

Comment:
No longer required since the new API shape allows a vector for the second operand, and therefore makes the upper two bits relevant.